### PR TITLE
Add more moves and implement field.isDefenseCurl

### PR DESCRIPTION
**Pull request by AustinXII**
Added isDefenseCurl to calculate the damage increase for moves like rollout and ice ball https://gyazo.com/4fc330aab9df9061a828658740439252

### CODE REVIEW

**Comment by AustinXII** [comment]
added everything that was requested with this commit: http://www.smogon.com/forums/threads/6th-gen-damage-calculator.3492947/page-6#post-5673022

File: calc_bc.html [hunk 1]
```diff
@@ -594,6 +594,13 @@
                             <input class="btn-input" type="checkbox" id="helpingHandR" />
                             <label class="btn btn-xxwide" for="helpingHandR">Helping Hand</label>
                         </div>
+                            <div class="left" title="Has this Pokemon's power been boosted by Defense Curl?">
+                                <input class="btn-input" type="checkbox" id="defenseCurlL" />
+                                <label class="btn btn-xxwide" for="defenseCurlL">Defense Curl</label>
+                        </div>
+                        <div class="right" title="Has this Pokemon's power been boosted by Defense Curl?">
+                            <input class="btn-input" type="checkbox" id="defenseCurlR" />
+                            <label class="btn btn-xxwide" for="defenseCurlR">Defense Curl</label>
```
Comment (Marty-D) [inline calc_bc.html]: The alignment and tag matching is still wrong here, should look like this: http://hastebin.com/hepasibixe.html

File: index.html [hunk 2]
```diff
@@ -633,6 +633,16 @@
                     <label class="btn btn-xxwide" for="helpingHandR">Helping Hand</label>
                 </div>
             </div>
+            <div class="btn-group gen-specific g3 g4 g5 g6">
```
Comment (Marty-D) [inline index.html]: Defense Curl doubled Rollout's power in Gen 2 as well.

File: js/ap_calc.js [hunk 3]
```diff
@@ -557,6 +559,8 @@ function clearField() {
     $("#foresightR").prop("checked", false);
     $("#helpingHandL").prop("checked", false);
     $("#helpingHandR").prop("checked", false);
+    $("#isDefenseCurlL").prop("checked", false);
+    $("#isDefenseCurlR").prop("checked", false);
```
Comment (Marty-D) [inline js/ap_calc.js]: `#defenseCurlL` and `#defenseCurlR` here.

File: js/damage.js [hunk 4]
```diff
@@ -315,6 +315,11 @@ function getDamageResult(attacker, defender, move, field) {
         description.isHelpingHand = true;
     }
     
+    if (field.isDefenseCurl && move.name === 'Ice Ball' || field.isDefenseCurl && move.name === 'Rollout') {
```
Comment (Marty-D) [inline js/damage.js]: `field.isDefenseCurl && (move.name === 'Ice Ball' || move.name === 'Rollout')`

File: js/damage.js [hunk 5]
```diff
@@ -593,6 +598,10 @@ function buildDescription(description) {
     if (description.isHelpingHand) {
         output += "Helping Hand ";
     }
+    output += description.attackerName + " ";
```
Comment (Marty-D) [inline js/damage.js]: The `attackerName` was already added to the display four lines up, so this line is unnecessary. Just add an extra space in `" Defense Curl "` two lines down.

File: js/damage_dpp.js [hunk 6]
```diff
@@ -193,6 +193,11 @@ function CALCULATE_DAMAGE_DPP(attacker, defender, move, field) {
         description.isHelpingHand = true;
     }
     
+    if (field.isDefenseCurl && move.name === 'Ice Ball' || field.isDefenseCurl && move.name === 'Rollout') {
+        bpMods.push(0x2000);
```
Comment (Marty-D) [inline js/damage_dpp.js]: `bpMods` doesn't exist in DPP or RSE.

File: js/data/move_data.js [hunk 7]
```diff
@@ -1074,6 +1094,12 @@ var MOVES_DPP = $.extend(true, {}, MOVES_ADV, {
         category: 'Physical',
         makesContact: true
     },
+    'Ominous Wind': {
+        bp:60,
```
Comment (Marty-D) [inline js/data/move_data.js]: Space!

**Comment by AustinXII** [comment]
Just putting this as a reminder to not merge until we fully finish how ice ball and rollout work

**Comment by AustinXII** [comment]
asdfasdfasdfasdf

**Comment by AustinXII** [comment]
@Marty-D  I was thinking of trying to do something along these lines for rollout and iceball http://hastebin.com/idufinabac.avrasm

**Comment by Zarel** [comment]
You should pay attention to your commit summary here, too.

**Comment by AustinXII** [comment]
fixed

**Comment by Marty-D** [comment]
@AustinXII, that seems fine, but I think the main problem with using `isMultiHit` for them is that their damage will be calculated as an average of the number of hits, like actual multi-hit moves are currently calculated. That doesn't seem like useful information to have, since this damage will actually be happening over multiple turns.

Assuming Rollout and Ice Ball are relevant, a new way of tracking hits should be implemented, which could then also be used for Fury Cutter, Echoed Voice, etc., as appropriate. `move.hitNum`? With an `isContinuous` flag for moves? Just spitballing some ideas.

**Comment by AustinXII** [comment]
@Marty-D that sounds like a good idea. I'll just move some of the moves I added on this pull request to the other one seeing how this one may be open for a while.

**Pull request closed without merging**